LSPS1/README.md

```diff
@@ -283,19 +284,78 @@ The client MAY check the current status of the order at any point.
 
 ### 3. Payment
 
-This section describes the `payment` object returned by `lsps1.create_order` and `lsps1.get_order`. The client MUST pay the `bolt11_invoice` OR the `onchain_address`. Using both methods MAY lead to the loss of funds.
+This section describes the `payment` object returned by `lsps1.create_order` and `lsps1.get_order`.
 
-> **Rationale** On-chain payments are required for payments with higher amounts, especially to push `client_balance_sat` to the client. On-chain payments are also useful to onboard new users to Lightining. On the other hand, Lightning payments are the preferred way to do payments because they are quick and easily refundable.
+```json
+"payment": {
+    "onchain": {...}.
+    "bolt11": {...}
+}
+```
+
+An LSP MAY omit payment options.
+> **Rationale**: E.g: The LSP might not support onchain payments.
+
+> **Rationale**:
+> Fees and expiry dates are defined on the level of the payment-option.
+> This allows an LSP to provide the cheapest service for each payment option.
```

Comment:
I am aware of @2Fast2BCn's [comment](https://github.com/BitcoinAndLightningLayerSpecs/lsp/issues/95#issuecomment-1976025005) but can you give me a concrete example on what variable would change so quickly within a payment option that would make the added complexity of multiple expiry times worth it? I can't think of any so far? Maybe onchain fees?

Reply:
The problem is that `lsps1` suffers a free-option problem for onchain feerates.

A client can use this to his advantage using the following procedure

```
A client might want to open a channel between now and 6 hours.
The client performs `lsps1.create_order` immediately.
The client performs `lsps1.create_order` after 6 hours.
The client pays for the cheapest order
```

When a drastic fee-rate occurs the server might be required to open the channel at a loss.
Accepting the order can incur a real risk to the LSP.

To mitigate the risk the server could either
1. make the order expire quickly
2. charge a sufficiently high fee

By putting the `expiry_time` in the `payment_option` we ensure that
- bolt11 payments can use (1)
- onchain payments can use (2)

Reply:
First, as an LSP providing an order with an expiry time of 6hrs, I can use an insurance model to cover such cases.
The LSP adds 1% fees on top of each onchain order. This 1% will be your insurance in case of a unexpected fee spike. If 1 out of 100 orders sees a fee spike, the accumulated insurance will cover these costs.

Second, I don't see any upside for the client to time the channel opening this way. It even has the downside of the client needing to wait for the channel open and needing to monitor the fee rate.

Third, why should an order be valid for such a long time? 1hr is more than sufficient IMO.